### PR TITLE
Update DG with all features

### PR DESCRIPTION
**Pull request by ryantzr1**
Tidy up all the features

### CODE REVIEW

**Review by victorpengmx — APPROVED** [review]
LGTM!

**Review by tayian — APPROVED** [review]
Thanks Ryan! LGTM, let's work towards a more comprehensive developer guide together

File: docs/diagrams/rankSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+@startuml
```
Comment (geraldngjx) [inline docs/diagrams/rankSequenceDiagram.puml:1]: Great addition of the UML diagram!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -440,6 +466,32 @@ The NotePersonCommand also handles cases where the provided person or company in
 
 <div style="page-break-after: always;"></div>
 
+## Ranking People by Priority Feature: `rank`
```
Comment (geraldngjx) [inline docs/DeveloperGuide.md:469]: Great addition of the new command additions such as "rank"

**Pull request merged**